### PR TITLE
Expect-style tests for parser

### PR DESCRIPTION
**Pull request by Lupus**
This PR suggests an expect-style framework for testing compiler library and includes some tests for demonstration of the approach.

Sorry for ugly expect output, my `Format`-fu is pretty low...

### CODE REVIEW

File: src/tests/expectation/dune [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+(rule
+ (with-stdout-to
+  tests.output
+  (run ./tests.exe)))
+
+(rule
+ (alias runtest)
+ (action
+  (diff tests.expected tests.output)))
```
Comment (c-cube) [inline src/tests/expectation/dune:14]: I think this is not needed if `tests.expected` is present, actually. Dune should automatically do it.

File: src/compilerlib/pb_parsing_parse_tree.ml [hunk 2]
```diff
@@ -164,3 +164,170 @@ type proto = {
...
+    message.id message.message_name
+    (pp_print_list
+       ~pp_sep:(fun ppf () -> fprintf ppf ";@,")
+       pp_message_body_content)
+    message.message_body
+
+let pp_rpc ppf rpc =
+  fprintf ppf
+    "{@[<2>@,\
+     rpc_name = \"%s\";@,\
```
Comment (c-cube) [inline src/compilerlib/pb_parsing_parse_tree.ml:278]:
```suggestion
     rpc_name = %S;@,\
```

**Comment by c-cube** [comment]
This looks very nice, and the `Format` usage is good!

File: src/tests/expectation/dune (hunk 1 above)
Comment (Lupus) [reply src/tests/expectation/dune:14]: My quick experiment suggests that this is necessary. Without the diff command, dune does nothing with `tests.output`.

**Comment by Lupus** [comment]
@c-cube I've improved output a bit, and fixed compilation issue on older OCaml version. Please take a look, should be ready to merge.

File: src/tests/expectation/dune (hunk 1 above)
Comment (c-cube) [reply src/tests/expectation/dune]: try with the new commit (modify the expected file to introduce a bug). It should fail.

**Comment by c-cube** [comment]
Indeed, the `executable` stanza is passive, but `test` adds a lot of default behaviors (such as checking the output against an expected file if present).

**Pull request merged**